src/mapblock.h

```diff
@@ -669,41 +669,36 @@ typedef std::vector<MapBlock*> MapBlockVect;
 
 inline bool objectpos_over_limit(v3f p)
 {
-	// MAP_BLOCKSIZE must be subtracted to avoid an object being spawned just
-	// within the map generation limit but in a block and sector that extend
-	// beyond the map generation limit.
-	// This avoids crashes caused by sector over limit in createSector().
-	const float object_limit = (MYMIN(MAX_MAP_GENERATION_LIMIT,
-		g_settings->getU16("map_generation_limit")) - MAP_BLOCKSIZE) * BS;
-	return (p.X < -object_limit
-		|| p.X > object_limit
-		|| p.Y < -object_limit
-		|| p.Y > object_limit
-		|| p.Z < -object_limit
-		|| p.Z > object_limit);
+	const float max_limit_bs = MAX_MAP_GENERATION_LIMIT * BS;
```

Comment:
I doubt it, it may even have been made static before now and was changed back, or i may be remembering something else that was wrongly made static.
It needs to be changeable between game sessions.